File: backend/src/Fix.Application/UseCases/Organizations/Commands/UpdateFinancialsCommand.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Application.Organizations.Dtos;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Organizations.Commands;

[RequireRole(RoleCodes.EditOrganization)]
public sealed record UpdateFinancialsCommand(
    Guid UserId,
    Guid OrganizationId,
    decimal? Cash,
    decimal? CreditLines,
    decimal? MonthlyFixedCost,
    decimal? NetDebt,
    decimal? Ebitda,
    decimal? UsdDebt,
    DateOnly? ReferenceDate)
    : ICommand<OrganizationSetupDto>, IOrganizationRequest;

File: frontend/bff/src/application/organizations/commands/update-financials.command.ts
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

export interface UpdateFinancialsCommand {
  context: RequestContext;
  cash: number | null;
  creditLines: number | null;
  monthlyFixedCost: number | null;
  netDebt: number | null;
  ebitda: number | null;
  usdDebt: number | null;
  referenceDate: string | null;
}
